utils/annotations.py

```diff
@@ -0,0 +1,421 @@
+        ad_expr: Optional[hl.expr.ArrayNumericExpression] = None,
+        adj_expr: Optional[hl.expr.BooleanExpression] = None
+) -> hl.expr.StructExpression:
+    """
+    Returns a struct expression with genotype quality histograms based on the arguments given (dp, gq, ad).
+
+    Notes
+    -----
+    - If `gt_expr` is provided, will return histograms for non-reference samples only as well as all samples.
+    - `gt_expr` is required for the allele-balance histogram, as it is only computed on het samples.
+    - If `adj_expr` is provided, additional histograms are computed using only adj samples.
+
+    :param CallExpression gt_expr: Entry expression containing genotype
+    :param NumericExpression gq_expr: Entry expression containing genotype quality
+    :param NumericExpression dp_expr: Entry expression containing depth
+    :param ArrayNumericExpression ad_expr: Entry expression containing allelic depth (bi-allelic here)
+    :param BooleanExpression adj_expr: Entry expression containing adj (high quality) genotype status
+    :return: Genotype quality histograms expression
+    :rtype: StructExpression
+    """
+    qual_hists = {}
+    if gq_expr is not None:
+        qual_hists['gq_hist'] = hl.agg.hist(gq_expr, 0, 100, 20)
+    if dp_expr is not None:
+        qual_hists['dp_hist'] = hl.agg.hist(dp_expr, 0, 100, 20)
+
+    if gt_expr is not None:
+        qual_hists= {
+            **{f'{qual_hist_name}_all': qual_hist_expr for qual_hist_name, qual_hist_expr in qual_hists.items()},
+            **{f'{qual_hist_name}_alt': hl.agg.filter(gt_expr.is_non_ref(), qual_hist_expr) for qual_hist_name, qual_hist_expr in qual_hists.items()}
+        }
+        if ad_expr is not None:
+            qual_hists['ab_hist_alt'] = hl.agg.filter(gt_expr.is_het(), hl.agg.hist(ad_expr[1] / hl.sum(ad_expr), 0, 1, 20))
+
+    else:
+        qual_hists = {f'{qual_hist_name}_all': qual_hist_expr for qual_hist_name, qual_hist_expr in qual_hists.items()}
+
+    if adj_expr is not None:
+        qual_hists.update({
+            f'{qual_hist_name}_adj': hl.agg.filter(adj_expr, qual_hist_expr) for qual_hist_name, qual_hist_expr in qual_hists.items()
+        })
+
+    return hl.struct(**qual_hists)
+
+
+def age_hists_expr(
+        adj_expr: hl.expr.BooleanExpression,
+        gt_expr: hl.expr.CallExpression,
+        age_expr: hl.expr.NumericExpression,
+        lowest_boundary: int = 30,
+        highest_boundary: int = 80,
+        n_bins: int = 10
+) -> hl.expr.StructExpression:
+    """
+    Returns a StructExpression with the age histograms for hets and homs.
+
+    :param BooleanExpression adj_expr: Entry expression containing whether a genotype is high quality (adj) or not
+    :param CallExpression gt_expr: Entry expression containing the genotype
+    :param NumericExpression age_expr: Col expression containing the sample's age
+    :param int lowest_boundary: Lowest bin boundary (any younger sample will be binned in n_smaller)
```

Comment:
I think the lower bound was based on the distribution of ages in gnomAD v2 (very few samples were younger). The upper one was somewhat similar with the added worry that if stratified further it may start be considered identifiable information just because so few people make it to say >100